Update how-to-open-a-pull-request.md (#14)
gatsbyjs/gatsby-es: docs/contributing/how-to-open-a-pull-request.md

---
title: Como abrir un Pull Request
---

Una gran parte de contribuir al open source es enviar cambios a un proyecto: mejoras al código fuente o tests, actualizaciones a contenido de documentos, inclusive errores tipográficos o hipervínculos rotos. Este documento cubrirá lo que necesitas saber para **abrir un pull request** en Gatsby.

## Qué es un Pull Request (PR)?

En caso de que no estés familiarizado, hete aquí como el equipo de GitHub [definen un pull request](https://help.github.com/en/articles/about-pull-requests):

> > Los Pull requests te permiten decir a otros acerca de cambios que has subido en una branch en un repositorio en GitHub. Una vez que un pull request es abierto, puedes discutir y revisar los cambios potenciales con colaboradores y agregar commits de seguimiento antes que tus cambios sean fusionados en la branch (o rama) de base.

Reviewer: ```suggestion
> Los Pull requests te permiten decir a otros acerca de cambios que has subido en una rama de un repositorio en GitHub. Una vez que un pull request es abierto, puedes discutir y revisar los cambios potenciales con colaboradores y agregar commits de seguimiento antes de que tus cambios sean incluidos en la rama de master (o base).
```

> Gatsby utiliza el proceso de PR para revisar y probar cambios antes que sean agregados al repositorio de GitHub de Gatsby. Cualquier persona puede abrir un pull request. El mismo proceso es utilizado por todos los contribuyentes, sea que ésta sea tu primera contribución al código abierto (open source) o que seas un miembro del core team de Gatsby.

Reviewer: ```suggestion
Gatsby utiliza el proceso de PR para revisar y probar cambios antes que sean agregados al repositorio de GitHub de Gatsby. Cualquier persona puede abrir un pull request. El mismo proceso es utilizado por todos los contribuyentes, sea que ésta sea tu primera contribución a código abierto (open source) o que seas un miembro del equipo principal de Gatsby.
```

> Cuando alguien quiere contribuir a Gatsby, abren una solicitud para _hacer un pull request_ (extraer) de su código hacia dentro del repo. Dependiendo del tipo de cambio hecho, los PRs son categorizados como de:

Reviewer: ```suggestion
Cuando alguien quiere contribuir a Gatsby, abren una solicitud para _hacer un pull request_ (extraer) de su código hacia dentro del repositorio. Dependiendo del tipo de cambio hecho, los PRs son categorizados como de:
```

- [Documentación](#documentation)
- [Código](#code-changes)

> - [Guías rápidas or Galería de sitios](#starters-or-site-showcase)

Reviewer: ```suggestion
- [Guías rápidas o Galería de sitios](#starters-or-site-showcase)
```

> - [Blog posts](#blog-posts)

Reviewer: ```suggestion
- [Entradas de Blog](#blog-posts)
```

Recomendaciones para diferentes tipos de contribuciones se encontrarán en esta guía asi como a través de los documentos de contribuciones.

> ## Cosas para saber antes de abrir un PR

Reviewer: ```suggestion
## Cosas a saber antes de abrir un PR
```

Típicamente recomendamos [abrir un issue](/contributing/how-to-file-an-issue/) antes de un pull request si no hay ya un issue para el problema que quieres solucionar. Esto ayuda a facilitar una discusión antes de decidir una implementación.

Para algunos cambios, como errores tipográficos o hipervínculos rotos, sería apropiado abrir un pequeño PR por si mismo. Esto es de algún modo subjetivo de modo que si tienes preguntas, [siéntete libre de consultarnos](/contributing/how-to-contribute/#not-sure-how-to-start-contributing).

> El equipo core de Gatsby usa un proceso de triaging delineado en [Manejando los Pull Requests](/contributing/managing-pull-requests/), si estás interesado en aprender más acerca de como ello funciona.

Reviewer: ```suggestion
El equipo principal de Gatsby usa un proceso de "triaging" delineado en [Manejando los Pull Requests](/contributing/managing-pull-requests/), si estás interesado en aprender más acerca de como funciona eso.
```

## Abriendo PRs en Gatsby

> Para cualquier tipo de cambio a archivos en el repo de Gatsby, puedes seguir los pasos siguientes. Asegúrate de chequear tips adicionales para contribuir a varias partes del repo más adelante en este doc, como de cambios de documentos, blog posts, guías iniciales o mejoras en código y tests.

Reviewer: ```suggestion
Para cualquier tipo de cambio a archivos en el repo de Gatsby, puedes seguir los siguientes pasos. Asegúrate de comprobar las recomendaciones adicionales para contribuir a varias partes del repo más adelante en este documento, como de cambios de documentos, entradas de blog, guías iniciales o mejoras en código y pruebas.
```

> Algunos PRs pueden ser realizados completamente desde la [UI de GitHub](https://help.github.com/en/articles/creating-a-pull-request), tal como ediciones a los archivos README o documentos.

Reviewer: ```suggestion
Algunos PRs pueden ser realizados completamente desde la [interfaz de usuario de GitHub](https://help.github.com/en/articles/creating-a-pull-request), tal como ediciones a los archivos README o documentos.
```

> Para testear cambios localmente frente al [sitio y archivos de proyecto de Gatsby](https://github.com/gatsbyjs/gatsby), puedes hacer un fork del repo e instalar partes de él para que corran en tu computadora localmente.

Reviewer: ```suggestion
Para probar cambios localmente frente al [sitio y archivos de proyecto de Gatsby](https://github.com/gatsbyjs/gatsby), puedes hacer un fork del repositorio e instalar partes de él para que corran en tu computadora localmente.
```

- [Hacer un fork y clonar el repositorio de Gatsby](/contributing/setting-up-your-local-dev-environment/#gatsby-repo-install-instructions).

> - Instalar [yarn](https://yarnpkg.com/) para traer las dependencias y hacer un build del proyecto.

Reviewer: ```suggestion
- Instalar [yarn](https://yarnpkg.com/) para instalar las dependencias y compilar el proyecto.
```

- Sigue las instrucciones para la parte del proyecto que deseas cambiar. (Ver secciones específicas abajo.)

> - [Crear una branch en Git](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging) para aislar tus cambios:

Reviewer: ```suggestion
- [Crear una rama en Git](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging) para aislar tus cambios:
```

  ```shell
  git checkout -b some-change
  ```

> - Una vez que tienes cambios en Git que quieres empujar, [agrégalos y crea un commit](https://help.github.com/en/articles/adding-a-file-to-a-repository-using-the-command-line). Para información on sobre como estructurar tus commits, revisa el doc [Manejando PRs](/contributing/managing-pull-requests/#commit-and-pr-title).

Reviewer: ```suggestion
- Una vez que tienes cambios en Git que quieres subir, [agrégalos y crea un commit](https://help.github.com/en/articles/adding-a-file-to-a-repository-using-the-command-line). Para información sobre como estructurar tus commits, revisa la guía de [Manejando PRs](/contributing/managing-pull-requests/#commit-and-pr-title).
```

>   - Usando un caracter de punto `.` agregará todos los archivos no seguidos todavía en el actual directorio y subdirectorios.

Reviewer: ```suggestion
  - Usando un carácter de punto `.` agregará todos los archivos no seguidos todavía en el actual directorio y subdirectorios.
```

  ```shell
  git add .
  ```
  - Usando una herramienta visual como [GitHub Desktop](https://desktop.github.com/) o [GitX](https://rowanj.github.io/gitx/) puede ayudar para elegir sobre qué archivos y lineas hacer un commit.

> - Committear código hará correr el linter automático que usa [Prettier](https://prettier.io). Para correr el linter manualmente, corre un script npm en el directorio base del proyecto:

Reviewer: ```suggestion
- Haciendo commit del código hará correr el linter automático que usa [Prettier](https://prettier.io). Para ejecutar el linter manualmente, ejecuta un script npm en el directorio base del proyecto:
```

  ```shell
  npm run format
  ```

> - Aplica todos los cambios de linting antes de empujar cambios [enmedando el commit previo](https://help.github.com/en/articles/changing-a-commit-message) o agregando un nuevo commit. Para más acerca de linting y pruebas, visita el doc de [Manejando PRs](/contributing/managing-pull-requests/#automated-checks).

Reviewer: ```suggestion
- Haz commit de todos los cambios del "linting" antes de subir cambios [enmendando el commit previo](https://help.github.com/en/articles/changing-a-commit-message) o agregando un nuevo commit. Para más acerca de "linting" y pruebas, visita la guía de [Manejando PRs](/contributing/managing-pull-requests/#automated-checks).
```

  ```shell
  git commit --amend
  ```

> - Empuja tus cambios a tu fork, asumiendo que está configurado como [`origin`](https://www.git-tower.com/learn/git/glossary/origin):

Reviewer: ```suggestion
- Sube tus cambios a tu fork, asumiendo que está configurado como [`origin`](https://www.git-tower.com/learn/git/glossary/origin):
```

  ```shell
  git push origin head
  ```

> - Para abrir un PR con tus cambios ante el repo de Gatsby, puedes usar el[UI de Pull Requests de GitHub](https://help.github.com/en/articles/creating-a-pull-request). Alternativamente, puedes usar la linea de comandos: recomendamos [hub](https://github.com/github/hub) para ellos.

Reviewer: ```suggestion
- Para abrir un PR con tus cambios ante el repo de Gatsby, puedes usar la [interfaz de usuario de Pull Requests de GitHub](https://help.github.com/en/articles/creating-a-pull-request). Alternativamente, puedes usar la linea de comandos: recomendamos [hub](https://github.com/github/hub) para ello.
```

### PRs de Documentación

> El sitio de documentación de Gatsby.js vive mayormente en los directorios [www](https://github.com/gatsbyjs/gatsby/tree/master/www) y [docs](https://github.com/gatsbyjs/gatsby/tree/master/docs) en GitHub, incluyendo docs y contenido de tutoriales. Hay también algunos [ejemplos en el repositorio de Gatsby](https://github.com/gatsbyjs/gatsby/tree/master/examples) referenciados en la documentación.

Reviewer: ```suggestion
El sitio de documentación de Gatsby.js vive mayormente en los directorios [www](https://github.com/gatsbyjs/gatsby/tree/master/www) y [docs](https://github.com/gatsbyjs/gatsby/tree/master/docs) en GitHub, incluyendo documentaciones y contenido de tutoriales. Hay también algunos [ejemplos en el repositorio de Gatsby](https://github.com/gatsbyjs/gatsby/tree/master/examples) referenciados en la documentación.
```

Pasos adicionales para PRs de docs:

> - Para cambios solamente de docs, considera usar `git checkout -b docs/some-change` o `git checkout -b docs-some-change`, ya que esto recortará el proceso de CI (o integración continua) y solo correrá tareas de linting.

Reviewer: ```suggestion
- Para cambios solamente de documentación, considera usar `git checkout -b docs/some-change` o `git checkout -b docs-some-change`, ya que esto recortará el proceso de CI (o integración continua) y solo ejecutará tareas de "linting".
```

> Más instrucciones pueden ser encontradas en la página de [contribuciones a docs](/contributing/docs-contributions/).

Reviewer: ```suggestion
Más instrucciones pueden ser encontradas en la página de [contribuciones a documentación](/contributing/docs-contributions/).
```

### Cambios de código

> Instrucciones para realizar cambios al código fuente de Gatsby, tests, materiales internos, APIs, paquetes, y más pueden ser encontrados en los docs de contribuciones en [estableciendo tu ambiente de desarrollo local](/contributing/setting-up-your-local-dev-environment/). Hay también detalles adicionales en la página de [Contribuciones de código](/contributing/code-contributions/).

Reviewer: ```suggestion
Instrucciones para realizar cambios al código fuente de Gatsby, pruebas, materiales internos, APIs, paquetes, y más pueden ser encontrados en la documentación de contribuciones en [estableciendo tu ambiente de desarrollo local](/contributing/setting-up-your-local-dev-environment/). Hay también detalles adicionales en la página de [Contribuciones de código](/contributing/code-contributions/).
```

> ### Guías iniciales o galerías de sitios

Reviewer: ```suggestion
### Guías iniciales o Galería de sitios
```

Hay páginas específicas acerca de cómo contribuir a varias partes del ecosistema Gatsby: :

- [Aplicaciones a la galería](/contributing/site-showcase-submissions/)
- [Librería de guías iniciales](/contributing/submit-to-starter-library/)

> ### Posts del blog

Reviewer: ```suggestion
### Entradas del Blog
```

> Para contribuir al blog de Gatsby, es necesario primero pasar tu idea de contenido por el equipo de Gatsby antes de enviarla. Para más información, refiérete a la página de [contribuciones al blog y sitio web](/contributing/blog-and-website-contributions/), incluyendo cómo proponer una idea y cómo configurar el blog para que corra localmente.

Reviewer: ```suggestion
Para contribuir al blog de Gatsby, es necesario primero pasar tu idea de contenido por el equipo de Gatsby antes de enviarla. Para más información, refiérete a la página de [contribuciones al blog y sitio web](/contributing/blog-and-website-contributions/), incluyendo cómo proponer una idea y cómo configurar el blog para que corra localmente.
```
```suggestion
Para contribuir al blog de Gatsby, es necesario primero pasar tu idea de contenido por el equipo de Gatsby antes de enviarla. Para más información, refiérete a la página de [contribuciones al blog y sitio web](/contributing/blog-and-website-contributions/), incluyendo cómo proponer una idea y cómo configurar el blog para que se ejecute localmente.
```

## Actualiza tu fork con los últimos cambios de Gatsby

El repo de Gatsby de Github es muy activo, así que es probable que necesites actualizar tu fork con los últimos cambios para hacer un merge de tu código. Esto requiere agregar a Gatsby como un [upstream remote](https://help.github.com/en/articles/configuring-a-remote-for-a-fork):

> - Establece la URL del repo de Gatsby's como una fuente remota. El nombre de la fuente remota es arbitrario; este ejemplo usa `upstream`.

Reviewer: ```suggestion
- Establece la URL del repositorio de Gatsby como una fuente remota. El nombre de la fuente remota es arbitrario; este ejemplo usa `upstream`.
```

  ```shell
  git remote set-url upstream git@github.com:gatsbyjs/gatsby.git
  ```

>   - _Nota: esta sintaxis [usa SSH y keys: también puedes usar `https`](https://help.github.com/en/articles/which-remote-url-should-i-use) y tu usuario/contraseña._

Reviewer: ```suggestion
  - _Nota: esta sintaxis [usa SSH y llaves de seguridad: también puedes usar `https`](https://help.github.com/en/articles/which-remote-url-should-i-use) con tu usuario/contraseña._
```

> - Puedes verificar el nombre del remote y URL en cualquier momento:

Reviewer: ```suggestion
- Puedes verificar el nombre del remoto y URL en cualquier momento:
```

  ```shell
  git remote -v
  ```
- Traer los últimos cambios desde Gatsby:
  ```shell
  git fetch upstream master
  ```

> - [En la branch](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging) que quieres actualizar, haz un merge de cualquier cambio desde Gatsby hacia tu fork:

Reviewer: ```suggestion
- [En la rama](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging) que quieres actualizar, haz un merge de cualquier cambio desde Gatsby hacia tu fork:
```

  ```shell
  git merge upstream master
  ```
  - Si hay [conflictos de merge](https://help.github.com/en/articles/resolving-a-merge-conflict-on-github), querrás encargarte de ellos para conseguir un merge limpio.

> - Una vez que tu branch está en estado funcional, empuja los cambios a tu fork:

Reviewer: ```suggestion
- Una vez que tu branch está en estado funcional, sube los cambios a tu fork:
```

  ```shell
  git push origin head
  ```

> Para más información sobre cómo trabajar con repositorios upstream, [visita los docs de GitHub](https://help.github.com/en/articles/configuring-a-remote-for-a-fork).

Reviewer: ```suggestion
Para más información sobre cómo trabajar con repositorios upstream, [visita la documentación de GitHub](https://help.github.com/en/articles/configuring-a-remote-for-a-fork).
```

> _**Nota:** como miembro del repositorio de Gatsby, tambien puedes clonarlo directamente en vez de hacer un fork, y empujar tus cambios a [branches de features](https://git-scm.com/book/en/v1/Git-Branching-Branching-Workflows)._

Reviewer: ```suggestion
_**Nota:** como miembro del repositorio de Gatsby, también puedes clonarlo directamente en vez de hacer un fork, y empujar tus cambios a [ramas de funcionalidades](https://git-scm.com/book/en/v1/Git-Branching-Branching-Workflows)._
```

## Recursos adicionales

- CSS Tricks: [Cómo contribuir a un proyecto open source](https://css-tricks.com/how-to-contribute-to-an-open-source-project/)
- [Creando un pull request](https://help.github.com/en/articles/creating-a-pull-request) from GitHub
- [Configurando un remoto para un fork](https://help.github.com/en/articles/configuring-a-remote-for-a-fork)

> - [Qué remoto debería usar?](https://help.github.com/en/articles/which-remote-url-should-i-use)

Reviewer: ```suggestion
- [¿Qué remoto debería usar?](https://help.github.com/en/articles/which-remote-url-should-i-use)
```

> - [Merging y Branching de Git](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging)

Reviewer: ```suggestion
- [Merge y ramas de Git](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging)
```

> - [Feature Branching y Workflows](https://git-scm.com/book/en/v1/Git-Branching-Branching-Workflows)

Reviewer: ```suggestion
- [Ramas por funcionalidad y flujos de trabajo](https://git-scm.com/book/en/v1/Git-Branching-Branching-Workflows)
```

- [Resolviendo conflictos de merge](https://help.github.com/en/articles/resolving-a-merge-conflict-on-github)

> - [Manejando Pull Requests](/contributing/managing-pull-requests/) del equipo core de Gatsby.

Reviewer: ```suggestion
- [Manejando Pull Requests](/contributing/managing-pull-requests/) del equipo principal de Gatsby.
```
